clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1292,8 +1296,12 @@ public void handleResponse(AbstractResponse response) {
                 reenqueue();
             } else if (error == Errors.COORDINATOR_LOAD_IN_PROGRESS || error == Errors.CONCURRENT_TRANSACTIONS) {
                 reenqueue();
-            } else if (error == Errors.TRANSACTIONAL_ID_AUTHORIZATION_FAILED ||
-                    error == Errors.CLUSTER_AUTHORIZATION_FAILED) {
+            } else if (error == Errors.CLUSTER_AUTHORIZATION_FAILED) {
+                log.error("Cluster authorization failure, transition the producer state back to uninitialized");
+                fail(error.exception());
+                // transition back to uninitialized state to enable client side retry
+                transitionTo(State.UNINITIALIZED);
+            } else if (error == Errors.TRANSACTIONAL_ID_AUTHORIZATION_FAILED) {
```

Comment:
I wonder if we should handle this case consistently. I cannot think of any reason not to.

Reply:
yeah that sounds right. it seems like you they should just configure the ACL.